R/write.R

```diff
@@ -1,5 +1,31 @@
+#' Write into a file
+#'
+#' Writes lines to a file, taking the union of what's already there, if
+#' anything, and some new lines. Note, there is no explicit promise about the
+#' line order. This function is designed to modify simple config files like
+#' `.Rbuildignore` and `.gitignore`.
+#'
+#' @param path Path to target file. It is created if it does not exist, but the
+#'   parent directory must exist.
+#' @param new_lines Character vector of lines to add, if not already present.
+#' @param quiet Logical. Whether to message about what is happening.
+#'
+#' @return Logical indicating whether a write occurred, invisibly.
+#' @export
+#' @keywords internal
+#'
+#' @examples
+#' tmp <- tempfile()
+#' write_union(tmp, letters[1:3])
+#' readLines(tmp)
+#' write_union(tmp, letters[1:5])
+#' readLines(tmp)
+#'
+#' ## clean up
+#' file.remove(tmp)
```

Comment:
I wonder if there's a non-awkward way to give this a name starting with `edit_`? That would make it more clearly a member of an existing family.

Reply:
Yeah. But perhaps `write_union()` and `write_utf()` should also be exported -- why export only `write_union()`? And then they make their own little family.

Reply:
write_utf8 definitely belongs elsewhere but seems good to export write_over